charts/ai-software-templates/chatbot/templates/application-gitops-job.yaml

```diff
@@ -12,7 +12,118 @@ spec:
+              }
+
+              cd /tmp || exit
+              wget https://github.com/cli/cli/releases/download/v"$GH_VERSION"/gh_"$GH_VERSION"_linux_amd64.tar.gz || exit
+              tar xzf gh_"$GH_VERSION"_linux_amd64.tar.gz
+              GH_PATH=$(pwd)/gh_"$GH_VERSION"_linux_amd64/bin
+              export PATH="$GH_PATH":"$PATH"
+              export SOURCE_REPO_APP_CONTENT_PATH="chatbot"
+              export SOURCE_TEKTON_REPO_CONTENT_PATH="pac/pipelineRuns"
+              export TEKTON_FILE_APP_NAME_REPLACEMENT="application_name_replace"
+              export TEKTON_FILE_APP_NAMESPACE_REPLACEMENT="application_namespace_replace"
+              export TEKTON_FILE_QUAY_ACCOUNT_REPLACEMENT="quay_account_replace"
+              echo APP_NAME is "$APP_NAME"
+              echo APP_NAMESPACE is "$APP_NAMESPACE"
+              echo GITHUB_ORG_NAME is "$GITHUB_ORG_NAME"
+              echo GITHUB_SOURCE_REPO is "$GITHUB_SOURCE_REPO"
+              echo GITHUB_TETKON_SOURCE_REPO is "$GITHUB_TETKON_SOURCE_REPO"
+              echo GITHUB_DEFAULT_BRANCH is "$GITHUB_DEFAULT_BRANCH"
+              echo QUAY_ACCOUNT_NAME is "$QUAY_ACCOUNT_NAME"
+
+              export GH_TOKEN=$GITHUB_TOKEN
+              export GH_REPO="github.com/$GITHUB_ORG_NAME/$APP_NAME"
+
+              export HOME=/tmp
+
+              gh auth login -p https
+              gh auth setup-git --hostname=github.com
+
+              gh repo view "$GH_REPO" --json id 2> /dev/null
+
+              return_code=$?
+
+              if [ $return_code == 0 ]; then
+                echo Repo "$GH_REPO" already exists, checking if contents are OK
+                setup_data
+                cd $HOME || exit
+                gh repo clone "$GH_REPO" existing -- --no-tags --branch="$GITHUB_DEFAULT_BRANCH" --single-branch || exit
+                same=$(diff existing/Containerfile "$APP_NAME"/Containerfile)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/chatbot_ui.py "$APP_NAME"/chatbot_ui.py)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/requirements.txt "$APP_NAME"/requirements.txt)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                same=$(diff existing/.tekton/docker-push.yaml "$APP_NAME"/.tekton/docker-push.yaml)
+                return_code=$?
+                if [ $return_code != 0 ]; then
+                  echo "$same"
+                  exit 1
+                fi
+                echo Repo "$GH_REPO" contents OK, exiting.
+                exit 0
+              else
+                gh repo create "$GH_REPO" --public --gitignore Python --license Apache-2.0 || exit
```

Comment:
Is this create instant or should we:
 - Introduce a delay or
 - Put the `gh repo view` in a delayed finite sized loop so it basically waits x amount of time for it to be created

Reply:
the create is synchronous

https://github.com/cli/cli/blob/trunk/pkg/cmd/repo/create/http.go#L227-L243

Reply:
i.e. the response from the lower level APIs includes the metadata from the created repo